packages/aws-cdk/lib/api/util/cloudformation/stack-status.ts

```diff
@@ -50,3 +50,28 @@ export class StackStatus {
     return this.name + (this.reason ? ` (${this.reason})` : '');
   }
 }
+
+/**
+ * A utility class to inspect CloudFormation stack detailed statuses.
+ *
+ * @see https://docs.aws.amazon.com/cli/latest/reference/cloudformation/describe-stacks.html
+ */
+export class StackDetailedStatus {
```

Comment:
Rather than make a new class, would it make more sense to have this be part of the existing StackStatus class?

And it would be great to add unit tests for these new functions :)

Reply:
DONE. Just refactored the existing class and add some unit tests.